src/modules/profile/profile.controller.ts

```diff
@@ -13,4 +20,16 @@ export class ProfileController {
     const user = req.user;
     return this.profileService.upsert(body, user.id);
   }
+
+  @Get()
+  @UseGuards(AuthGuard('jwt'))
+  async getProfile(@Request() req) {
+    const user = req.user;
+    const profile = await this.profileService.findOne(user.id);
+    if (profile === undefined || profile === null) {
+      return { message: 'Profile not found' };
+    } else {
+      return profile;
+    }
```

Comment:
Again, let's keep these checks or validations in service, and you can throw an error 404 when a profile is not found. I recommend using the CustomNotFoundException to keep a pattern in the application.

Reply:
done.